src/net.h

```diff
@@ -980,28 +980,9 @@ class CConnman
     /**
      * Generate a collection of sockets to check for IO readiness.
      * @param[in] nodes Select from these nodes' sockets.
-     * @param[out] recv_set Sockets to check for read readiness.
-     * @param[out] send_set Sockets to check for write readiness.
-     * @param[out] error_set Sockets to check for errors.
-     * @return true if at least one socket is to be checked (the returned set is not empty)
+     * @return sockets to check for readiness
      */
-    bool GenerateSelectSet(const std::vector<CNode*>& nodes,
-                           std::set<SOCKET>& recv_set,
-                           std::set<SOCKET>& send_set,
-                           std::set<SOCKET>& error_set);
-
-    /**
-     * Check which sockets are ready for IO.
-     * @param[in] nodes Select from these nodes' sockets.
-     * @param[out] recv_set Sockets which are ready for read.
-     * @param[out] send_set Sockets which are ready for write.
-     * @param[out] error_set Sockets which have errors.
-     * This calls `GenerateSelectSet()` to gather a list of sockets to check.
-     */
-    void SocketEvents(const std::vector<CNode*>& nodes,
-                      std::set<SOCKET>& recv_set,
-                      std::set<SOCKET>& send_set,
-                      std::set<SOCKET>& error_set);
+    Sock::EventsPerSock GenerateWaitSockets(const std::vector<CNode*>& nodes);
```

Comment:
Instead of `const std::vector<CNode*>&` could use a `std::span<const CNode*>` which is more general?

Reply:
Leaving it as is - I tried to make it `Span`, but it can't be to `const CNode*` because the method does `LOCK(pnode->cs_vSend);` for each `CNode` element. This then results in the error: `No viable conversion from 'const std::vector<CNode *>' to 'Span<CNode *>'`.

Reply:
Alright, having it `Span<CNode* const>` works (not `Span<const CNode*>`)! The pointer itself has to be constant - should not be able to re-point it to another `CNode` object. And that pointer is pointing to a non-constant `CNode` object, so that we can modify it (or at least lock its member mutex which is equivalent to modifying). Changed!

PS `std::span` is a `C++20` thing, we are using our own `Span` while on `C++17`.

Reply:
Great!

> PS std::span is a C++20 thing, we are using our own Span while on C++17.

Eh, yes, true.